File: targets/gitignore_test.go
package targets

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestGitignoreMatch(t *testing.T) {
	assert := assert.New(t)
	matched := false

	matched, _ = GitignoreMatch(".git", ".git/message")
	assert.True(matched)

	matched, _ = GitignoreMatch("node_modules", "node_modules/x/message.ts")
	assert.True(matched)

	matched, _ = GitignoreMatch("**/.git", ".git/message")
	assert.True(matched)

	matched, _ = GitignoreMatch("/.git", ".git/message")
	assert.True(matched)

	matched, _ = GitignoreMatch("message", ".git/message")
	assert.True(matched)

	matched, _ = GitignoreMatch("/message", ".git/message")
	assert.False(matched)
}

File: src/patterns/gitignore.test.ts
import { ok } from "node:assert/strict";
import { test } from "node:test";
import { gitignoreMatch } from "./gitignore.js";

void test("gitignoreMatch", () => {
  ok(gitignoreMatch(".git", ".git/message"));
  ok(!gitignoreMatch(".git", ".github/message"));

  ok(gitignoreMatch("node_modules", "node_modules/x/message.ts"));
  ok(gitignoreMatch("message", ".git/message"));

  ok(gitignoreMatch("**/.git", ".git/message"));
  ok(!gitignoreMatch("**/.git", ".github/message"));

  ok(gitignoreMatch("/.git", ".git/message"));
  ok(!gitignoreMatch("/.git", ".github/message"));

  ok(!gitignoreMatch("/message", ".git/message"));
  ok(!gitignoreMatch("/message", ".git/message/file"));
  ok(gitignoreMatch("/message", "message"));
  ok(gitignoreMatch("/message", "message/file"));

  ok(gitignoreMatch(".git/", ".git/message"));
  ok(gitignoreMatch(".git/", ".git/message/file"));
  ok(gitignoreMatch(".git/", ".git"));
  ok(!gitignoreMatch(".git/", ".github/message"));
  ok(!gitignoreMatch(".git/", ".github/message/file"));
  ok(!gitignoreMatch(".git/", ".github"));
});
